pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/BrokerRegistry.java

```diff
@@ -0,0 +1,100 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.broker.loadbalance.extensions;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.BiConsumer;
+import org.apache.pulsar.broker.PulsarServerException;
+import org.apache.pulsar.broker.loadbalance.extensions.data.BrokerLookupData;
+import org.apache.pulsar.metadata.api.NotificationType;
+
+/**
+ * Responsible for registering the current Broker lookup info to
+ * the distributed store (e.g. Zookeeper) for broker discovery.
+ */
+public interface BrokerRegistry {
+
+    /**
+     * Start broker registry.
+     */
+    void start();
+
+    /**
+     * Register local broker to metadata store.
+     */
+    void register();
+
+    /**
+     * Unregister the broker.
+     *
+     * Same as {@link org.apache.pulsar.broker.loadbalance.ModularLoadManager#disableBroker()}
+     */
+    void unregister() throws PulsarServerException;
+
+    /**
+     * Get the current broker lookup service address.
+     *
+     * @return The service url without the protocol prefix, 'http://'. e.g. broker-xyz:port
+     */
+    String getLookupServiceAddress();
+
+    /**
+     * Get available brokers.
+     *
+     * @return The brokers service url list.
+     */
+    List<String> getAvailableBrokers();
+
+    /**
+     * Async get available brokers.
+     *
+     * @return The brokers service url list.
+     */
+    CompletableFuture<List<String>> getAvailableBrokersAsync();
+
+    /**
+     * Fetch local-broker data from load-manager broker cache.
+     *
+     * @param broker The service url without the protocol prefix, 'http://'. e.g. broker-xyz:port
+     */
+    Optional<BrokerLookupData> lookup(String broker);
+
+    /**
+     * For each the broker lookup data.
+     * The key is lookupServiceAddress{@link BrokerRegistry#getLookupServiceAddress()}
+     */
+    void forEach(BiConsumer<String, BrokerLookupData> action);
+
+    /**
+     * Listen the broker register change.
+     *
+     * @param listener Key is lookup service address{@link BrokerRegistry#getLookupServiceAddress()}
+     *                 Value is notification type.
+     */
+    void listen(BiConsumer<String, NotificationType> listener);
+
+    /**
+     * Close the broker registry.
+     *
+     * @throws PulsarServerException if it fails to close the broker registry.
+     */
+    void close() throws PulsarServerException;
```

Comment:
Should we just extends `AutoCloseable` for all interfaces that have `close` method?